File: testdata/polyglot/fp/webgaps.go
// Safe-code plants for the FP measurement eval.
package fp

import (
	"crypto/tls"
	"net/http"
	"os"
	"path/filepath"
)

func safeSSRF() (*http.Response, error) {
	// PLANT-FP(go-safe-ssrf, CWE-918): constant URL, no user-controlled host.
	return http.Get("https://api.example.com/status")
}

func safeRead(base, in string) ([]byte, error) {
	// PLANT-FP(go-safe-path, CWE-22): filepath.Base strips directory components.
	return os.ReadFile(filepath.Join(base, filepath.Base(in)))
}

func safeTLS() *tls.Config {
	// PLANT-FP(go-safe-tls, CWE-295): verification left enabled.
	return &tls.Config{MinVersion: tls.VersionTLS13}
}

File: testdata/polyglot/fp/webgaps.js
// Safe-code plants for the FP measurement eval.
const axios = require("axios");
const fs = require("fs");
const path = require("path");

// PLANT-FP(js-safe-ssrf, CWE-918): constant, hard-coded URL.
const a = axios.get("https://api.internal.example.com/status");

// PLANT-FP(js-safe-path, CWE-22): basename strips directory traversal.
const b = fs.readFileSync(path.basename(req.query.file));

// PLANT-FP(js-safe-redirect, CWE-601): constant redirect target.
function go(res) { res.redirect("/dashboard"); }
